tests/test_actions.py

```diff
@@ -640,3 +654,181 @@ def test_describe(self) -> None:
+
+    def test_performable_performs(self, Tester):
+        action = FakeAction()
+        Quietly(action).perform_as(Tester)
+        action.perform_as.assert_called_once_with(Tester)
+
+    def test_resolvable_resolves(self):
+        resolution = FakeResolution()
+        Quietly(resolution).resolve()
+        resolution.resolve.assert_called_once_with()
+
+
+class Action1(Performable):
+    @beat("{} tries to Action1")
+    def perform_as(self, actor: Actor):
+        actor.will(Quietly(Action2()))
+
+
+class Action2(Performable):
+    @beat("{} tries to Action2")
+    def perform_as(self, actor: Actor):
+        settings.DEBUG_QUIETLY = True
+        actor.will(Quietly(See(SimpleQuestion(), IsEqualTo(True))))
+
+
+class Action3(Performable):
+    @beat("{} tries to Action3")
+    def perform_as(self, actor: Actor):
+        actor.will(Quietly(Action4()))
+
+
+class Action4(Performable):
+    @beat("{} tries to Action4")
+    def perform_as(self, actor: Actor):
+        actor.will(Quietly(See(SimpleQuestion(), IsEqualTo(True))))
+
+
+class TestQuietlyDebug:
+    @pytest.fixture(autouse=True)
+    def setup(self):
+        settings.DEBUG_QUIETLY = False
+        yield
+        settings.DEBUG_QUIETLY = False
+
+    def test_kinking(self, Tester: Actor, mocker: MockerFixture):
+        mock_clear = mocker.spy(the_narrator, "clear_backup")
+        mock_flush = mocker.spy(the_narrator, "flush_backup")
+        mock_kink = mocker.spy(the_narrator, "mic_cable_kinked")
+
+        Tester.will(Quietly(FakeAction()))
+
+        assert mock_kink.call_count == 1
+        assert mock_clear.call_count == 2
+        assert mock_flush.call_count == 1
+
+    def test_skip_creation(self) -> None:
+        settings.DEBUG_QUIETLY = True
+        q = Quietly(FakeAction())
+        assert isinstance(q, FakeAction)
+
+    def test_debug_no_kink(self, Tester: Actor, mocker: MockerFixture) -> None:
+        mocker.patch.object(settings, "DEBUG_QUIETLY", return_value=True)
+        mock_clear = mocker.spy(the_narrator, "clear_backup")
+        mock_flush = mocker.spy(the_narrator, "flush_backup")
+        mock_kink = mocker.spy(the_narrator, "mic_cable_kinked")
+
+        Quietly(FakeAction()).perform_as(Tester)
+        assert mock_kink.call_count == 0
+        assert mock_clear.call_count == 0
+        assert mock_flush.call_count == 0
+
+    def test_debug_logging(self, Tester: Actor, mocker: MockerFixture) -> None:
+        mocker.patch.object(settings, "DEBUG_QUIETLY", return_value=True)
+        mock_clear = mocker.spy(the_narrator, "clear_backup")
+        mock_flush = mocker.spy(the_narrator, "flush_backup")
+        mock_kink = mocker.spy(the_narrator, "mic_cable_kinked")
+
+        Tester.will(QuietlyPerformable(FakeAction()))
+        assert mock_kink.call_count == 1
+        assert mock_clear.call_count == 1
+        assert mock_flush.call_count == 1
+
+    def test_realtime_1(self, Tester, caplog) -> None:
```

Comment:
Can this name be a little more descriptive? I'm not sure what "test realtime 1" or its siblings "test realtime 2" and "test realtime 3" mean.

Reply:
It's `TestQuietlyDebug::test_realtime_1`.  It could be all in one test method called `test_realtime` but that felt odd since there are 3 variations of that test.  I couldn't think of a better descriptor for each test variation without writing a sentence.  
I'm happy to change it.